handler/handler.go

```diff
@@ -145,8 +155,7 @@ func (h *ReindexRequestedHandler) getAndSendDatasetURLs(ctx context.Context, cfg
 	datasetChan := h.extractDatasets(ctx, &wgDataset, datasetAPICli, cfg.ServiceAuthToken)
 	editionChan := h.retrieveDatasetEditions(ctx, &wgDataset, datasetAPICli, datasetChan, cfg.ServiceAuthToken)
 	datasetURLChan := h.getAndSendDatasetURLsFromLatestMetadata(ctx, &wgDataset, datasetAPICli, editionChan, cfg.ServiceAuthToken)
-	urlCount := h.logExtractedDatasetURLs(ctx, &wgDataset, cfg, datasetURLChan, reindexReqEvent)
-	// TODO - logExtractedDatasetURLs is temporary and should be replaced in the future
+	urlCount := h.sendExtractedDatasetURLs(ctx, &wgDataset, cfg, datasetURLChan, reindexReqEvent)
```

Comment:
In conjunction with other comment further down this file… wait before this function and make the function sequential rather than concurrent.
```suggestion
        wgDataset.Wait() // wait for the other go-routines to complete which extracts the dataset urls
	urlCount := h.sendExtractedDatasetURLs(ctx, cfg, datasetURLChan, reindexReqEvent)
```
~```--- wgDataset.Wait() // wait for the other go-routines to complete which extracts the dataset urls```~